docs/guides/go-built-in.md

```diff
@@ -0,0 +1,630 @@
+# 1 Guide Assumptions
+
+This guide is designed for beginners who want to get started with a Tendermint
+Core application from scratch. It does not assume that you have any prior
+experience with Tendermint Core.
+
+Tendermint Core is Byzantine Fault Tolerant (BFT) middleware that takes a state
+transition machine - written in any programming language - and securely
+replicates it on many machines.
+
+Although Tendermint Core is written in the Golang programming language, prior
+knowledge of it is not required for this guide. You can learn it as we go due
+to it's simplicity. However, you may want to go through [Learn X in Y minutes
+Where X=Go](https://learnxinyminutes.com/docs/go/) first to familiarize
+yourself with the syntax.
+
+By following along with this guide, you'll create a Tendermint Core project
+called kvstore, a (very) simple distributed BFT key-value store.
+
+# 1 Creating a built-in application in Go
+
+Running your application inside the same process as Tendermint Core will give
+you the best possible performance.
+
+For other languages, your application have to communicate with Tendermint Core
+through a TCP, Unix domain socket or gRPC.
+
+## 1.1 Installing Go
+
+Please refer to [the official guide for installing
+Go](https://golang.org/doc/install).
+
+Verify that you have the latest version of Go installed:
+
+```sh
+$ go version
+go version go1.12.7 darwin/amd64
+```
+
+Make sure you have `$GOPATH` environment variable set:
+
+```sh
+$ echo $GOPATH
+/Users/melekes/go
+```
+
+## 1.2 Creating a new Go project
+
+We'll start by creating a new Go project.
+
+```sh
+$ mkdir -p $GOPATH/src/github.com/me/kvstore
```

Comment:
this isn't required anymore with go modules, do we still want to include it?

Reply:
🙈 Tendermint requires GOPATH for some of its scripts. Better to leave it for now